test/test_cli.rb

```diff
@@ -9,34 +9,31 @@ class TestCLI < Minitest::Test
 
   def setup
     @environment = 'production'
-    @tmp_file = Tempfile.new("puma-test")
-    @tmp_path = @tmp_file.path
-    @tmp_file.close!
-
-    @tmp_path2 = "#{@tmp_path}2"
+    @tmp_path  = "tmp/#{full_file_name}"
+    @tmp_path2 = "#{@tmp_path}_2"
 
     File.unlink @tmp_path  if File.exist? @tmp_path
     File.unlink @tmp_path2 if File.exist? @tmp_path2
 
     @wait, @ready = IO.pipe
 
     @events = Puma::Events.strings
-    @events.on_booted { @ready << "!" }
+    @events.on_booted { @ready.syswrite '!' }
   end
 
   def wait_booted
     @wait.sysread 1
   end
 
   def teardown
-    File.unlink @tmp_path if File.exist? @tmp_path
+    File.unlink @tmp_path  if File.exist? @tmp_path
     File.unlink @tmp_path2 if File.exist? @tmp_path2
 
-    @wait.close
-    @ready.close
+    @ready.close rescue nil
```

Comment:
I think you can probably understand why I don't think just adding `rescue nil` everywhere is fixing anything, right? 😆

Reply:
It's in a test.  If something else goes wrong before it...